plugins/destination/duckdb/client/write.go

```diff
@@ -115,6 +116,54 @@ func (c *Client) copyFromFile(ctx context.Context, tableName string, fileName st
 		") from '"+fileName+"' (FORMAT PARQUET)")
 }
 
+func (c *Client) appendRows(ctx context.Context, table *schema.Table, msgs message.WriteInserts) (retErr error) {
+	defer func() {
+		if retErr != nil {
+			fmt.Println("===", retErr.Error())
+		}
+	}()
+	if c.conn == nil {
+		// connecting before MigrateTable results in appender creation failure
+		var err error
+		c.conn, err = c.connector.Connect(context.Background())
+		if err != nil {
+			return fmt.Errorf("failed to connect: %w", err)
+		}
+	}
+
+	appender, err := duckdb.NewAppenderFromConn(c.conn, "", table.Name)
+	if err != nil {
+		return fmt.Errorf("failed to create appender for %s: %w", table.Name, err)
+	}
+	defer func() {
+		err := appender.Close()
+		if err != nil && retErr == nil {
+			retErr = fmt.Errorf("failed to close appender for %s: %w", table.Name, err)
+		}
+	}()
+
+	tableState, err := c.getCacheTableInfo(ctx, table.Name)
+	if err != nil {
+		return fmt.Errorf("table %s not found in duckdb: %w", table.Name, err) // should never happen as appender would've failed
+	}
+
+	fields := msgs[0].Record.Schema().Fields()
+
+	for _, msg := range msgs {
+		arr, err := transformRecordToGoType(msg.Record, fields, tableState.Columns)
```

Comment:
`appender.AppendRow` gets a `driver.Value` so we need to convert Arrow records to Go types.